src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2169,23 +2182,8 @@ private String launchCommandIn(ArgumentListBuilder args, File workDir, EnvVars e
 
                 status = process.joinWithTimeout(usedTimeout, TimeUnit.MINUTES, listener);
 
-                StringBuilder stdoutBuilder = new StringBuilder();
-                try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getStdout(), encoding))) {
-                    String line;
-                    while ((line = reader.readLine()) != null) {
-                        stdoutBuilder.append(line);
-                    }
-                }
-                stdout = stdoutBuilder.toString();
-
-                StringBuilder stderrBuilder = new StringBuilder();
-                try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getStderr(), encoding))) {
-                    String line;
-                    while ((line = reader.readLine()) != null) {
-                        stderrBuilder.append(line);
-                    }
-                }
-                stderr = stderrBuilder.toString();
+                stdout = getZosStdio(process, encoding, false);
```

Comment:
Why not `InputStream` as parameter?
```
stdout = getZosStdio(process.getStdout(), encoding);
stderr = getZosStdio(process.getStderr(), encoding);
```
The method is independent of ZOS. Therefore the name could be something like `readStreamIntoString()`.

There are even existing implementations for this - see https://stackoverflow.com/questions/309424/how-to-read-convert-an-inputstream-into-a-string-in-java
```
stdout = IOUtils.toString(process.getStdout(), encoding);
stderr = IOUtils.toString(process.getStderr(), encoding);
```

Reply:
After more careful reading of your comments @darxriggs , I believe I've understood your concerns and implemented something that uses your suggestion.